ilc/client/GuardManager.js

```diff
@@ -0,0 +1,66 @@
+import errors from '../common/guard/errors';
+import actionTypes from '../common/guard/actionTypes';
+
+export default class GuardManager {
+    #router;
+    #transitionHooksPlugin;
+    #errorHandler;
+
+    constructor(router, pluginManager, errorHandler) {
+        this.#router = router;
+        this.#transitionHooksPlugin = pluginManager.getTransitionHooksPlugin();
+        this.#errorHandler = errorHandler;
+    }
+
+    hasAccessTo(url) {
+        if (this.#transitionHooksPlugin === null) {
+            return true;
+        }
+
+        const route = this.#router.match(url);
+
+        if (route.specialRole !== null) {
+            return true;
+        }
+
+        const hooks = this.#transitionHooksPlugin.getTransitionHooks();
+
+        if (hooks.length === 0) {
+            return true;
+        }
+
+        for (const hook of hooks) {
+            try {
+                const action = hook({
+                    route: {...route, url},
+                    navigate: this.#router.navigateToUrl,
+                });
+
+                if (action.type === actionTypes.stopNavigation) {
+                    return false;
```

Comment:
Add here log to console: `ILC: Stopped navigation due to the Route Guard with index #XXX`

Reply:
Updated